src/Microsoft.Diagnostics.EventFlow.Inputs.MicrosoftLogging/Logger/EventFlowLoggerScope.cs

```diff
@@ -0,0 +1,75 @@
+﻿// ------------------------------------------------------------
+//  Copyright (c) Microsoft Corporation.  All rights reserved.
+//  Licensed under the MIT License (MIT). See License.txt in the repo root for license information.
+// ------------------------------------------------------------
+
+using System;
+#if NET451
+using System.Runtime.Remoting.Messaging;
+#else
+using System.Threading;
+#endif
+
+namespace Microsoft.Diagnostics.EventFlow.Inputs
+{
+    public class EventFlowLoggerScope
+    {
+        internal EventFlowLoggerScope(object state)
+        {
+            State = state;
+        }
+
+        public Object State { get; private set; }
+        public EventFlowLoggerScope Parent { get; private set; }
+
+#if NET451
+        private static readonly string FieldKey = $"{typeof(EventFlowLoggerScope).FullName}";
```

Comment:
this code is error prone for RPC calls.
1. EventFlowLoggerScope is not Serializable and does not inherit from MarshalRefByObject, which means that once cross appdomain call is done, app will crash with some SerializationException.
2. Even if you implement one or another way, second appdomain may have no knowledge about event flow - that is a typical case for app running in IIS. So when the second appdomain will initialize, app will crash not being able to resolve EventFlowLoggerScope 
3. Additionally, there are some issues with mutability of the object, potential copy during serialization, etc. so cross appdomain support is really complex.

So I suggest following (basically prevent things to be reused in the other app-domains):
1. Make FieldKey app-domain specific - add AppDomain.Current.Id to it
2. EventFlowLoggerScope wrap EventFlowLoggerScope  into ObjectHandle before string in the Logical call context.

This way, it will be successfully serialized, other appdomain will be happy to find ObjectHandle in the logical call context (will resolve the type), however no one will ever attempt to access EventFlowLoggerScope  from the other app-domain, so no actual resolution of EventFlowLoggerScope type, access and modification of object will happen.

This is the lesson learned from the Activity in the hard way :)